src/marqo/tensor_search/tensor_search.py

```diff
@@ -1314,14 +1319,28 @@ def search(config: Config, index_name: str, text: Optional[Union[str, dict]],
         if approximate is None:
             approximate = True
 
+        if search_method.upper() == SearchMethod.TENSOR:
+            search_result = _vector_text_search(
+                config=config, index_name=index_name, query=text, result_count=result_count, offset=offset,
+                ef_search=ef_search, approximate=approximate, searchable_attributes=searchable_attributes,
+                filter_string=filter, device=selected_device, attributes_to_retrieve=attributes_to_retrieve,
+                boost=boost,
+                image_download_headers=image_download_headers, context=context, score_modifiers=score_modifiers,
+                model_auth=model_auth, highlights=highlights, text_query_prefix=text_query_prefix
+            )
+        elif search_method.upper() == SearchMethod.HYBRID:
+            # TODO: Deal with circular import when all modules are refactored out.
```

Comment:
What is the import circle here?

Reply:
It's because `core/search/hybrid_search.py` uses a lot of infra from `tensor_search.py`. The way this could be fixed in the future is if most of the functionality was split out into separate modules. Right now I get:
```
from marqo.tensor_search.tensor_search import run_vectorise_pipeline, gather_documents_from_response, logger
ImportError: cannot import name 'run_vectorise_pipeline' from partially initialized module 'marqo.tensor_search.tensor_search' (most likely due to a circular import) (/Users/joshua/work/marqo_main/marqo/src/marqo/tensor_search/tensor_search.py)
```